coordinator/src/notification_service.rs

```diff
@@ -130,3 +144,205 @@ async fn send_notification<'a>(
     tracing::debug!("Sent: {:?}", response);
     Ok(())
 }
+
+/// Send notifications to users with positions that are about to expire or have
+/// just expired
+pub async fn send_expiry_notifications_if_applicable(
+    positions: &[(Position, FcmToken)],
+    notification_sender: &mpsc::Sender<Notification>,
+) {
+    let now = OffsetDateTime::now_utc();
+
+    for (position, fcm_token) in positions {
+        if position.expiry_timestamp <= now
+            && now < position.expiry_timestamp + time::Duration::hours(1)
```

Comment:
:question: Can we use a `Range` to define this predicate? It should be more readable if possible.

Reply:
cool idea! I'll look into it!

Reply:
there's no ranges support for `OffsetDateTime` unfortunately :(